arrows/core/applets/track_features.cxx

```diff
@@ -0,0 +1,501 @@
+      if( !mdv.empty() )
+      {
+        converted_image->set_metadata( mdv[0] );
+      }
+
+      // Load the mask for this image if we were given a mask image list
+      kwiver::vital::image_container_sptr mask, converted_mask;
+      if( use_masks )
+      {
+        mask = image_reader->load( mask_files[ts.get_frame()] );
+
+        // error out if we are not expecting a multi-channel mask
+        if( !expect_multichannel_masks && mask->depth() > 1 )
+        {
+          LOG_ERROR( main_logger,
+                     "Encounted multi-channel mask image!" );
+          return EXIT_FAILURE;
+        }
+        else if( expect_multichannel_masks && mask->depth() == 1 )
+        {
+          LOG_WARN( main_logger,
+                    "Expecting multi-channel masks but received one that was "
+                    "single-channel." );
+        }
+
+        if( invert_masks )
+        {
+          LOG_DEBUG( main_logger,
+                     "Inverting mask image pixels" );
+          kwiver::vital::image_of<bool> mask_image;
+          kwiver::vital::cast_image( mask->get_image(), mask_image );
+          kwiver::vital::transform_image( mask_image, [] (bool b) { return !b; } );
+          LOG_DEBUG( main_logger,
+                     "Inverting mask image pixels -- Done" );
+          mask = std::make_shared<kwiver::vital::simple_image_container>( mask_image );
+        }
+
+        converted_mask = image_converter->convert( mask );
+      }
+
+      tracks = feature_tracker->track(tracks, ts.get_frame(),
+                                      converted_image, converted_mask);
+      if (tracks)
+      {
+        tracks = kwiver::arrows::core::extract_feature_colors(tracks, *image, ts.get_frame());
+      }
+
+      // Compute ref homography for current frame with current track set + write to file
+      // -> still doesn't take into account a full shotbreak, which would incur a track reset
+      if ( homog_ofs.is_open() )
+      {
+        LOG_DEBUG(main_logger, "writing homography");
+        homog_ofs << *(out_homog_generator->estimate(ts.get_frame(), tracks)) << std::endl;
+      }
+    }
+
+    if ( homog_ofs.is_open() )
+    {
+      homog_ofs.close();
+    }
+
+    // Writing out tracks to file
+    kwiver::vital::write_feature_track_file(tracks, output_tracks_file);
+
+    return EXIT_SUCCESS;
+  }
+  catch (std::exception const& e)
+  {
+    LOG_ERROR(main_logger, "Exception caught: " << e.what());
+
+    return EXIT_FAILURE;
+  }
+  catch (...)
+  {
+    LOG_ERROR(main_logger, "Unknown exception caught");
+
+    return EXIT_FAILURE;
+  }
+} // run
+
+// ----------------------------------------------------------------------------
+void
+track_features::
+add_command_options()
+{
+  m_cmd_options->custom_help( wrap_text( "[options]\n" ) );
+
+  m_cmd_options->add_options()
+    ( "h,help",     "Display applet usage" )
+    ( "c,config",   "Configuration file for tool", cxxopts::value<std::string>() )
```

Comment:
In a future branch, I would like to add options for input and output files like "input-video" and "output-tracks".  The "config" option would then be optional.  We would have a default config file for this applet in the `config` directory that would be loaded automatically if the "config" option is not specified.  The input and output variables would override the corresponding values in the default or loaded config.  This would let you run this tool like this:
```
kwiver track_features my_video.mp4 tracks.txt
```
without any messing with configuration files.